packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -61,5 +278,48 @@ protected override MethodProvider[] BuildMethods()
 
             return methods.ToArray();
         }
+
+        private ParameterProvider[] GetClientParameters()
+        {
+            var parameterCount = _inputClient.Parameters.Count;
+            ParameterProvider[] clientParameters = new ParameterProvider[parameterCount];
+
+            for (var i = 0; i < parameterCount; i++)
+            {
+                var inputClientParam = _inputClient.Parameters[i];
+                var parameterProvider = new ParameterProvider(inputClientParam);
+
+                if (inputClientParam.IsEndpoint)
+                {
+                    // add the endpoint parameter
+                    var knownEndpointParam = KnownParameters.Endpoint;
+                    var description = inputClientParam.Description ?? $"{knownEndpointParam.Description}";
+                    var endpointValue = parameterProvider.InitializationValue != null
+                        ? New.Instance(knownEndpointParam.Type, parameterProvider.InitializationValue)
+                        : null;
+                    var endpointParam = new ParameterProvider(
+                        knownEndpointParam.Name,
+                        $"{description}",
+                        knownEndpointParam.Type,
+                        initializationValue: endpointValue)
+                    {
+                        Validation = parameterProvider.Validation
+                    };
+
+                    clientParameters[i] = endpointParam;
+                }
+                else
+                {
+                    clientParameters[i] = parameterProvider;
+                }
+            }
+
+            return clientParameters;
+        }
+
+        private FieldProvider? GetFieldByConstructorParameterName(string parameterName)
```

Comment:
shouldn't be needed since a param has a field property

Reply:
removed.